### PR TITLE
Add Unit Tests to Loklak Server

### PR DESCRIPTION
**Pull request by vibhcool**
Presently Loklak Server has no Automated Unit Testing 

Pre-Settings:

I have added a profile @loklak_test to twitter for testing of TwitterScraper.java

Changes : 
1) commit: Add unit tests for TwitterScraper.java
 - creates unit test for TwitterScraper. (more tests needed to be added, working on it)
 - creates TestRunner which can run all tests added to it.
 - modifies TwitterScraper.java so that it can be tested. i.e. change all private access modifier to public access modifier.

2) commit: Add data file to test JSONRandomAccessFileTest.java
 - fixes existing test by adding datafile to be tested

3) commit: Set up unit tests build in loklak Server
 - fixes:  issue #883
 - add test building settings in build.gradle
 - create test.sh to run tests (TestRunner.java)

Steps to run unit tests :-

Step 1 : ./gradlew test
Step 2 : bin/test.sh

### CODE REVIEW

**Comment by vibhcool** [comment]
Tests are passing and open for review :)

**Comment by vibhcool** [comment]
@yukiisbored on my local machine, the project is building with no error and all the tests are showing passed on my machine.
the error shown is in the existing file, this pr hadn't touched.  
Could you try this pr on your machine?

**Comment by vibhcool** [comment]
@yukiisbored @Orbiter 
edited pr details, fixed error, tested and open for review :)

File: bin/test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env sh
+cd $(dirname $0)/..
+
+# Execute preload script
+source bin/.preload.sh
+
+echo "starting Tests"
+
+java -Xmx1024m -classpath .:build/libs/loklak_server-all.jar:build/classes/test org.junit.runner.JUnitCore org.loklak.TestRunner
```
Comment (yukiisbored) [inline bin/test.sh]: Euhm, to be quite honest just use `gradlew tests` instead of using the JUnit Test Runner.

File: test/org/loklak/tools/storage/messages_20170411_17092725.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{"timestamp":"2017-04-11T16:20:21.827Z","created_at":"2017-04-11T16:20:05.000Z","screen_name":"SouliVincent","text":"Courage madame le maire il faut nettoyer autour proprement car vous êtes entouré de vieux délinquant dealers de parcelle","link":"https://twitter.com/SouliVincent/status/851832235686866944","id_str":"851832235686866944","canonical_id":"","parent":"","source_type":"TWITTER","provider_type":"SCRAPED","retweet_count":0,"favourites_count":0,"place_name":"Volx","place_id":"","user":{"appearance_first":"2017-04-11T16:20:21.827Z","profile_image_url_https":"https://abs.twimg.com/sticky/default_profile_images/default_profile_bigger.png","screen_name":"SouliVincent","user_id":"841578791348645888","name":"Souli vincent@yahoo","appearance_latest":"2017-04-11T16:20:21.827Z"}}
```
Comment (yukiisbored) [inline test/org/loklak/tools/storage/messages_20170411_17092725.txt]: Can you decrease this into the needed ones instead of a huge example?

File: test/org/loklak/harvester/TwitterScraperTest.java [hunk 3]
```diff
@@ -73,13 +68,14 @@ public void test_search() {
...
+        }
 
+        ftweet_list = processTweetList(tweet_list);
         //compare no. of tweets with fetched no. of tweets
-        assertEquals(2, this.ftweet_list.size());
+        assertEquals(2, ftweet_list.size());
     }
 
-    public Timeline process_tweet_list(Timeline[] tweet_list) {
+    public Timeline processTweetList(Timeline[] tweet_list) {
```
Comment (yukiisbored) [inline test/org/loklak/harvester/TwitterScraperTest.java]: I prefer if this is on another commit/PR.

File: test/org/loklak/harvester/TwitterScraperTest.java [hunk 4]
```diff
@@ -62,8 +58,7 @@ public void test_search() {
 
                 //fetch list of tweets and set in ftweet_list
                 tweet_list = TwitterScraper.search(br, order, true, true);
-                this.ftweet_list = process_tweet_list(tweet_list);
-
+
```
Comment (yukiisbored) [inline test/org/loklak/harvester/TwitterScraperTest.java]: Useless spaces.

File: test/org/loklak/harvester/TwitterScraperTest.java [hunk 5]
```diff
@@ -43,8 +39,8 @@ public void test_prepareSearchURL() {
     }
 
     @Test
-    public void test_search() {
```
Comment (yukiisbored) [inline test/org/loklak/harvester/TwitterScraperTest.java]: Is this really needed by JUnit? If not, please make it into a separate commit because this is completely unrelated to this change.

File: bin/test.sh (hunk 1 above)
Comment (vibhcool) [inline bin/test.sh]: gradle compiles the junit test files and run them to check test files. Where as bin/test.sh just run the tests to test the project files.

File: test/org/loklak/harvester/TwitterScraperTest.java (hunk 5 above)
Comment (vibhcool) [inline test/org/loklak/harvester/TwitterScraperTest.java]: I added 2 methods in TwitterScraper to just create a test it to initialize automated tests. More tests are in pipeline (working on them) when this pr gets merged.

Comment (vibhcool) [inline test/org/loklak/harvester/TwitterScraperTest.java]: done, made all unrelated changes in separate commit :)

File: test/org/loklak/harvester/TwitterScraperTest.java (hunk 3 above)
Comment (vibhcool) [inline test/org/loklak/harvester/TwitterScraperTest.java]: done, made this change in another commit :)

File: test/org/loklak/tools/storage/messages_20170411_17092725.txt (hunk 2 above)
Comment (vibhcool) [inline test/org/loklak/tools/storage/messages_20170411_17092725.txt]: reduced the datafile from 41 lines to 5. Please see :)

**Comment by vibhcool** [comment]
@yukiisbored made the requested changes, please review :)

**Comment by yukiisbored** [comment]
A lot of unnecessary commits for a single purpose, please squash it.

**Comment by vibhcool** [comment]
@yukiisbored i have added last commit to fix the errors and changes.
 The rest commits are for specific portions of code. like:- 
1) 'Add unit tests for TwitterScraper.java' for adding twitterscrapertest, 
2) 'Set up unit tests build in loklak Server' for initializing testing structure in loklak server
3) 'Add data file to test JSONRandomAccessFileTest.java' to fix existing test in loklak server

**Comment by vibhcool** [comment]
thank you :)
